src/data-context.js

```diff
@@ -38,54 +39,96 @@ export function DataProvider({children}) {
             "professional"
         ]),
         color: faker.commerce.color()
-        })
-    );
-    
+    }));
+
+    function manipulateQuantity( state, {manipulation, item}){
+        switch(manipulation){
+
+            case "ADD_TO_CART":
+                {
+                    item = {...item, quantity: 1}
+                    return {...state, itemsInCart: [...state.itemsInCart, item]}
+                }
+            
+            case "INCREASE":
+                return { ...state, itemsInCart : [
+                    ...state.itemsInCart.map((cartItem) => cartItem.id === item.id ?  {...cartItem, quantity: cartItem.quantity + 1} : cartItem)
+                ]}
+
+            case "DECREASE":
+                return { ...state, itemsInCart : [
+                    ...state.itemsInCart.map((cartItem) => cartItem.id === item.id ?  {...cartItem, quantity: cartItem.quantity - 1} : cartItem)
+                ]}
+
+            case "REMOVE":
```

Comment:
you can change the payload name to `REMOVE_FROM_CART` for clear understanding of what payload is doing.